modules/transactions/src/main/java/org/apache/ignite/internal/tx/storage/state/TxStateStorage.java

```diff
@@ -63,7 +64,7 @@ public interface TxStateStorage extends AutoCloseable {
      * @throws IgniteInternalException with {@link Transactions#TX_STATE_STORAGE_ERR} error code in case when
      *                                 the operation has failed.
      */
-    boolean compareAndSet(UUID txId, TxState txStateExpected, TxMeta txMeta, long commandIndex);
+    boolean compareAndSet(UUID txId, @Nullable TxState txStateExpected, TxMeta txMeta, long commandIndex);
```

Comment:
Again, please specify the meaning of `null`. Is this documented somewhere?

Reply:
Probably, the authors of `PartitionListener` should be asked, as `PartitionListener` always pass `null` here as the second argument. I just noted this and added an annotation the make this visible.

Reply:
Yes, the guys say that it's ok. This means that there was no state before the operation

Reply:
Didn't know that, thank you!